src/card/block.spec.js

```diff
@@ -0,0 +1,180 @@
+
+        test( '🔬 Text Richtext editor should be editable.', async () => {
+            await blockStartup(block)
+
+            const typeString = "Test text here."
+            
+            // focus the textarea then type into it with the virtual keyboard
+            await page.focus( '.block-editor-rich-text__editable.content' );
+            await page.keyboard.type(typeString, {delay: 10})
+
+            const editedContent = await getEditedPostContent()
+
+            // tests: 
+            // a. save output properly contains the correct string 
+            // b. save output matches the existing snapshot
+            expect( editedContent.includes(`<p class="content">${typeString}</p>`)).toBe(true)
+            expect( editedContent).toMatchSnapshot()
+        })
+    })
+    describe( '🔬 Side Panel Settings', () => {
+        test( '🔬 Gold background color should be selectable.', async () => {
+            await blockStartup(block)
+
+            // select the radio button with the virtual keyboard
+            await clickRadio('Background Color', 'gold')
+
+            const editedContent = await getEditedPostContent()
+
+            expect( editedContent.includes(`"backgroundColor":"white"`)).toBe(false)
+            expect( editedContent).toMatchSnapshot()
+        })
+        test( '🔬 Black border color should be selectable.', async () => {
+            await blockStartup(block)
+        
+            // select the radio button with the virtual keyboard
+            await clickRadio('Border Color', 'black')
+
+            const editedContent = await getEditedPostContent()
+
+            expect( editedContent.includes(`"borderColor":"gold"`)).toBe(false)
+            expect( editedContent).toMatchSnapshot()
+        })
+        test( '🔬 Image alt text should be editable.', async () => {
+            await blockStartup(block)
+
+            const typeString = "Image alt text."
+            
+            // focus the text box then type into it with the virtual keyboard
+            await clickElementByText('label', 'Image Alt Text')
+            await page.keyboard.type(typeString, {delay: 10})
+
+            const editedContent = await getEditedPostContent()
+
+            // tests: 
+            // a. save output properly contains the correct string 
+            // b. save output matches the existing snapshot
+            expect( editedContent.includes(`"altText":"${typeString}"`)).toBe(true)
+            // expect( editedContent).toMatchSnapshot()
+        })
+        describe( '🔬 Link can be added to the card', () => {
+            test( '🔬 "Add a link to this card?" checkbox reveals link inputs', async () => {
+                await blockStartup(block)
+
+                await clickCheckbox('Add a link to this card?')
+        
+                const linkInputs = await page.$x(
+                    `//label[@class = "components-checkbox-control__label"][.="Call to action text" or .="Link address" or .="Open link in new tab?"]`
```

Comment:
this is only selecting the checkbox control label, not the other 2 textboxes that show up.

Reply:
You could change the XPath query to just `//label[.="Call to action text" or .="Link address" or .="Open link in new tab?"]`
That way it avoids looking for a label with a specific class, and only looks for labels with those 3 text options.

Reply:
Ah, I thought it also tests the show-up of the text boxes at the  beginning and later I forgot to update it.